src/Symfony/Bridge/Twig/Extension/TranslationExtension.php

```diff
@@ -104,17 +105,24 @@ public function getTranslationNodeVisitor(): TranslationNodeVisitor
     }
 
     /**
-     * @param ?string|Translatable $message The message id (may also be an object that can be cast to string)
+     * @param string|\Stringable|TranslatableInterface|null $message
+     * @param array|string                                  $arguments Can be the locale as a string when $message is a TranslatableInterface
      */
-    public function trans($message, array $arguments = [], string $domain = null, string $locale = null, int $count = null): string
+    public function trans($message, $arguments = [], string $domain = null, string $locale = null, int $count = null): string
```

Comment:
The new semantics of the `$arguments` parameter is not covered by a test, is it? I also wonder if this change is really necessary. I find the error message rather confusing that a developer gets when passing an array of arguments to the `trans` filter (which can be an honest C&P mistake).

Reply:
Test case added as `{{ t("Hello")|trans("fr") }}`

Not doing this polymorphism means that ppl would be forced to always write
`{{ translateble|trans(locale=fr) }}` instead of `{{ translateble|trans(fr) }}` which the current changes allows.

If the issue is the error message, we could improve it. Suggestions welcome :)